Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -232,29 +236,41 @@ BTDiagnostics::ReadParameters ()
     if (utils::parser::queryWithParser(pp_diag_name, "buffer_size", m_buffer_size)) {
         if(m_max_box_size < m_buffer_size) m_max_box_size = m_buffer_size;
     }
-
+#ifdef WARPX_DIM_RZ
+    pp_diag_name.query("dump_rz_modes", m_dump_rz_modes);
+    amrex::Vector< std::string > BTD_varnames_supported = {"Er", "Et", "Ez",
+                                                           "Br", "Bt", "Bz",
+                                                           "jr", "jt", "jz", "rho"};
+#else
+    amrex::ignore_unused(m_dump_rz_modes);
     amrex::Vector< std::string > BTD_varnames_supported = {"Ex", "Ey", "Ez",
                                                            "Bx", "By", "Bz",
                                                            "jx", "jy", "jz", "rho"};
+#endif
 
     for (const auto& var : m_varnames) {
         WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
             (utils::algorithms::is_in(BTD_varnames_supported, var )),
             "Input error: field variable " + var + " in " + m_diag_name
             + ".fields_to_plot is not supported for BackTransformed diagnostics."
             + " Currently supported field variables for BackTransformed diagnostics "
-            + "include Ex, Ey, Ez, Bx, By, Bz, jx, jy, jz, and rho");
+            + "include Ex, Ey, Ez, Bx, By, Bz, jx, jy, jz, and rho in Cartesian coordinates and "
+            + "Er, Et, Ez, Br, Bt, Bz, jr, jt, jz, and rho in cylindrical (RZ coordinates)");
     }
 
     bool particle_fields_to_plot_specified = pp_diag_name.queryarr("particle_fields_to_plot", m_pfield_varnames);
     WARPX_ALWAYS_ASSERT_WITH_MESSAGE(!particle_fields_to_plot_specified, "particle_fields_to_plot is currently not supported for BackTransformed Diagnostics");
+    if (m_varnames.size() == 0) {
+        m_do_back_transformed_fields = false;
+    }
 
 }
 
 bool
 BTDiagnostics::DoDump (int step, int i_buffer, bool force_flush)
 {
     // timestep < 0, i.e., at initialization time when step == -1
+    amrex::Print() << " buffer " << i_buffer << " full ? " << buffer_full(i_buffer) << " snapshot full " << m_snapshot_full[i_buffer] << " empty ? " << buffer_empty(i_buffer) << "\n";
```

Comment:
Leftover debug statement?
```suggestion
    // amrex::AllPrint() << " buffer " << i_buffer << " full ? " << buffer_full(i_buffer) << " snapshot full " << m_snapshot_full[i_buffer] << " empty ? " << buffer_empty(i_buffer) << "\n";
```